torch/csrc/jit/ir.cpp

```diff
@@ -425,40 +380,42 @@ void Graph::lint() const {
   // - uniques in all_nodes are unique
   // - every use will occur later in the topsort
 
-  std::unordered_set<const Node*> in_scope;
+  std::unordered_set<const Value*> in_scope;
+  std::unordered_set<const Node*> node_in_scope;
   std::unordered_set<size_t> seen_uniques;
   std::unordered_map<const Node*, int64_t> anticipated_uses;
+  auto check_value = [&](const Value* v) {
+    auto b = in_scope.insert(v);
+    JIT_ASSERT(b.second);  // insertion took place
+    auto b2 = seen_uniques.insert(v->unique());
+    JIT_ASSERT(b2.second);  // insertion took place
+    JIT_ASSERT(v->unique() < next_unique_);
+
+    for (auto use : v->uses()) {
+      JIT_ASSERT(node_in_scope.count(use.user) == 0);
+      JIT_ASSERT(all_nodes.count(use.user) == 1);
+      anticipated_uses[use.user]++;  // int default constructs to 0
+    }
+  };
   auto check_node = [&](const Node* n) {
     for (auto input : n->inputs_) {
       if (in_scope.count(input) != 1) {
-        if (n->kind_ == kSelect) {
-          JIT_ASSERTM(0, "%%%d (select node) not in scope; you probably forget to append it to the graph (you won't see this in the graph rendering)", input->unique_);
-        } else {
-          JIT_ASSERTM(0, "%%%d not in scope", input->unique_);
-        }
+        JIT_ASSERTM(0, "%%%d not in scope", input->unique());
       }
     }
     JIT_ASSERT(anticipated_uses[n] == static_cast<int64_t>(n->inputs_.size()));
     anticipated_uses[n] = -1;  // we saw the anticipated user!
-
-    auto b = in_scope.insert(n);
-    JIT_ASSERT(b.second);  // insertion took place
-    auto b2 = seen_uniques.insert(n->unique_);
-    JIT_ASSERT(b2.second);  // insertion took place
-    JIT_ASSERT(n->unique_ < next_unique_);
-
-    for (auto use : n->uses_) {
-      JIT_ASSERT(in_scope.count(use.user) == 0);
-      JIT_ASSERT(all_nodes.count(use.user) == 1);
-      anticipated_uses[use.user]++;  // int default constructs to 0
+    auto node_inserted = node_in_scope.insert(n);
+    JIT_ASSERT(node_inserted.second);  // insertion took place
+    for(auto o : n->outputs()) {
+      check_value(o);
```

Comment:
We should `JIT_ASSERT(o->node() == n)` here (previously this assert was handled by uses, but there are no uses of a node anymore, so we need to restore this assert.)

Reply:
Also, let's check that `offset_` is consistent with the ordering in `outputs()`. This will have the added bonus of ensuring there are not duplicate Value in the IR.

Reply:
(These checks could also easily go in `check_value`; since this is the only call site of `check_value` it is equivalent to do it either way.)